clippy_dev/src/update_lints.rs

```diff
@@ -14,6 +14,43 @@ const GENERATED_FILE_COMMENT: &str = "// This file was generated by `cargo dev u
 
 const DOCS_LINK: &str = "https://rust-lang.github.io/rust-clippy/master/index.html";
 
+macro_rules! declare_lint_crates {
+    ($($name:ident,)*) => {
+        #[allow(non_camel_case_types)]
+        #[derive(Debug, Clone, Copy, PartialEq, Eq)]
+        pub enum Crate {
+            $($name),*
+        }
+        impl Crate {
+            #[must_use]
+            pub fn src_path(self) -> &'static str {
+                match self {
+                    $(Self::$name => concat!(stringify!($name), "/src")),*
+                }
+            }
+
+            fn crate_root_path(self) -> &'static str {
```

Comment:
NIT: Maybe a better name `crate_lib_path` or even more explicit `crate_lib_rs_path`.

No strong opinion here though.

Reply:
I'll leave this for when I end up having to rebase. Hopefully I don't forget.